buildSrc/src/toolkits/gradle/changelog/ChangeLogGenerator.kt

```diff
@@ -0,0 +1,84 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package toolkits.gradle.changelog
+
+import org.gradle.api.logging.Logging
+import java.nio.file.Path
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
+import kotlin.streams.toList
+
+/* ktlint-disable custom-ktlint-rules:log-not-lazy */
+/**
+ * Generates a combined change log file based in Markdown syntax
+ */
+class ChangeLogGenerator(private val writers: List<ChangeLogWriter>) {
```

Comment:
Now that you're chaining this do you really need the multiple `writers` ?

Reply:
We still only generate certain ones at certain times so I think so